STUDIO/org.openl.rules.webstudio/src/org/openl/rules/webstudio/web/repository/RepositoryTreeState.java

```diff
@@ -811,9 +823,18 @@ private boolean isMainBranch(UserWorkspaceProject selectedProject) {
         return mainBranch;
     }
 
+    private boolean isCurrentBranchProtected(UserWorkspaceProject selectedProject) {
+        Repository designRepository = selectedProject.getDesignRepository();
+        if (designRepository.supports().branches()) {

```

Comment:
designRepository can be null for Local projects. Could you add check for null for safety reasons?